src/Pool.sol

```diff
@@ -587,140 +533,98 @@ abstract contract Pool {
         public
         view
-        returns (bytes32 weightsHash)
+        returns (bytes32 receiversHash)
     {
-        return subSenders[senderAddr][subSenderId].receiverWeightsHash;
+        return subSenders[senderAddr][subSenderId].receiversHash;
     }
 
-    function _weightsSum(ReceiverWeight[] calldata receiverWeights)
+    /// @notice Calculates the total amount per second of all the passed receivers.
+    /// @param receivers The list of the receivers.
+    /// @return totalAmtPerSec The total amount per second
+    function _totalAmtPerSec(Receiver[] calldata receivers)
         internal
         pure
-        returns (uint32 weightsSum)
+        returns (uint128 totalAmtPerSec)
     {
-        for (uint256 i = 0; i < receiverWeights.length; i++) {
-            weightsSum += receiverWeights[i].weight;
+        for (uint256 i = 0; i < receivers.length; i++) {
+            totalAmtPerSec += receivers[i].amtPerSec;
         }
     }
 
-    /// @notice Sets the weight of the provided receivers of the user.
-    /// The weight regulates the share of the amount sent every second
-    /// that each of the sender's receivers gets.
-    /// Setting a non-zero weight for a new receiver adds it to the list of the sender's receivers.
-    /// Setting zero as the weight for a receiver removes it from the list of the sender's receivers.
+    /// @notice Sets the receivers of the sender.
     /// @param sender The updated sender
-    /// @param newReceivers The list of the user's receivers and their weights,
-    /// which shall be in use after this function is called.
-    function _setReceivers(Sender storage sender, ReceiverWeight[] calldata newReceivers) internal {
-        require(newReceivers.length <= SENDER_WEIGHTS_COUNT_MAX, "Too many receivers");
-        uint256 senderWeightSum = 0;
-        for (uint256 i = 0; i < newReceivers.length; i++) {
-            senderWeightSum += newReceivers[i].weight;
-            require(newReceivers[i].weight > 0, "Receiver weight zero");
+    /// @param receivers The new list of the user's receivers
+    function _setReceivers(Sender storage sender, Receiver[] calldata receivers) internal {
+        require(receivers.length <= MAX_RECEIVERS, "Too many receivers");
+        uint256 totalAmtPerSec = 0;
+        for (uint256 i = 0; i < receivers.length; i++) {
+            require(receivers[i].amtPerSec != 0, "Receiver amtPerSec is zero");
+            totalAmtPerSec += receivers[i].amtPerSec;
             if (i > 0) {
-                address prevReceiver = newReceivers[i - 1].receiver;
-                address currReceiver = newReceivers[i].receiver;
+                address prevReceiver = receivers[i - 1].receiver;
+                address currReceiver = receivers[i].receiver;
                 require(prevReceiver <= currReceiver, "Receivers not sorted by address");
                 require(prevReceiver != currReceiver, "Duplicate receivers");
             }
         }
```

Comment:
I think this might be cheaper, if we reduce the lines of code in the loop
```solidity
        require(receivers[0].receiver != 0, "Receiver amtPerSec is zero");
        uint256 totalAmtPerSec = receivers[0].receiver.amtPerSec;
        for (uint256 i = 1; i < receivers.length; i++) {
            require(receivers[i].amtPerSec != 0, "Receiver amtPerSec is zero");
            totalAmtPerSec += receivers[i].amtPerSec;
            address prevReceiver = receivers[i - 1].receiver;
            address currReceiver = receivers[i].receiver;
            require(prevReceiver <= currReceiver, "Receivers not sorted by address");
            require(prevReceiver != currReceiver, "Duplicate receivers");
        }
```

Reply:
We'd cut out a single `if` on a local variable in a loop, I really don't think that it's worth the code duplication